File: buildSrc/src/main/java/net/twisterrob/inventory/database/Category.java
package net.twisterrob.inventory.database;

class Category {
	static final int INVALID_ID = Integer.MIN_VALUE;
	Category parent;
	String name;
	int id = INVALID_ID;
	int level = 0;
	String icon;

	@Override public String toString() {
		return "Category{" +
				"name='" + name + '\'' +
				", id=" + id +
				", level=" + level +
				", icon='" + icon + '\'' +
				'}';
	}
	public void setParent(Category parent) {
		this.parent = parent;
		if (icon == null && parent != null) {
			icon = parent.icon;
		}
	}
}

File: gradle/plugins-inventory/src/main/kotlin/net/twisterrob/inventory/database/Category.kt
package net.twisterrob.inventory.database

class Category {

	var parent: Category? = null
		set(value) {
			field = value
			if (icon == null && value != null) {
				icon = value.icon
			}
		}

	var name: String? = null
	var id = INVALID_ID
	var level = 0
	var icon: String? = null

	override fun toString(): String {
		return "Category{" +
			"name='" + name + '\'' +
			", id=" + id +
			", level=" + level +
			", icon='" + icon + '\'' +
			'}'
	}


	companion object {

		const val INVALID_ID = Int.MIN_VALUE
	}
}
